### PR TITLE
support syslog module

### PR DESCRIPTION
**Pull request by anas-shami**
support syslog module to config. logging levels and priorities 

You can configure: 
- local level 
- trap 
- format 
- monitor facility and override class
- enable receive logging 
- change fraction digits or whole digits in the seconds field

### CODE REVIEW

File: lib/ansible/modules/network/onyx/onyx_syslog.py [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+    enabled:
+        description:
+          - Enable/Disable logging properity
+        choices: ['enabled', 'disabled']
+        type: str
+        default: 'enabled'
+    local_level:
+        description:
+          - Set minimum severity of log messages saved on the local disk
+        choices: ['alert', 'crit', 'debug', 'emerg', 'err', 'info', 'none', 'notice', 'warning']
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_syslog.py]: please sort this list from least severe to most severe

File: lib/ansible/modules/network/onyx/onyx_syslog.py [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+            override_class:
+                description:
+                - Specify a class whose log level to override
+                choices: ['mgmt-front', 'mgmt-back', 'mgmt-core', 'events', 'debug-module', 'sx-sdk', 'mlx-daemons', 'protocol-stack']
+                type: str
+                required: true
+            override_priority:
+                description:
+                - Specify a priority whose log level to override
+                choices: ['alert', 'crit', 'debug', 'emerg', 'err', 'info', 'none', 'notice', 'warning']
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_syslog.py]: use sorted list.
please use the sorted list in all relevant places

File: lib/ansible/modules/network/onyx/onyx_syslog.py [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+                            'override_class': override_class,
+                            'override_priority': override_priority
+                        })
+                    else:
+                        current_config['local_override'] = [{
+                            'override_class': override_class,
+                            'override_priority': override_priority
+                        }]
+
+        current_config['received'] = True if current_config.get('received') == 'yes' else False
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_syslog.py]:
```python
current_config['received'] = (current_config.get('received') == 'yes')
```

File: lib/ansible/modules/network/onyx/onyx_syslog.py [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+        ''' extract 'format/fields' from running config line (if any ?) its single line - no'''
+        if syslog_config and syslog_config[0].get('lines'):
+            lines = syslog_config[0].get('lines')
+            for line in lines:
+                match = self.FORMAT_WELF_RGX.match(line)
+                if match:
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_syslog.py:309]:
```python
if match:
    current_config['format_welf_host'] = match.group(1)
    continue
```
no need to proceed to other checks when you find a match
do it for all match check

**Review by samerd — CHANGES_REQUESTED** [review]
Thanks @anas-shami 
please fix according to the comments

**Comment by anas-shami** [comment]
@samerd  please review

**Review by samerd — APPROVED** [review]
looks great
thanks @anas-shami

**Comment by samerd** [comment]
@ganeshrn 
could you please review and approve this PR if it is ok?

**Comment by Akasurde** [comment]
Hi @anas-shami, Thank you very much for your interest in Ansible. This plugin/module is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/mellanox.onyx<br><br>If you have further questions please stop by IRC or the mailing list:<br><br>* IRC: #ansible on irc.freenode.net<br>* mailing list: https://groups.google.com/forum/#!forum/ansible-project<br><br>needs_info<br>

**Pull request closed without merging**